File: OCH_Tao/ABC/C/337.py
#未AC
N = int(input())
A = list(map(int,input().split()))
l = [A.index(-1)+1]
for i in range(N-1):
  x = l[i]
  l.append(A.index(x)+1)
print(" ".join(list(map(str,l))))

File: OCH_Tao/ABC/C/337.js
//未AC
const input = (require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n"));
const N = parseInt(input[0], 10);
const A = (input[1].split(" ")).map(x => parseInt(x, 10));
let l = [A.indexOf(-1) + 1];
for (let i = 0; i < N - 1; i++) {
  let x = l[i];
  l.push(A.indexOf(x) + 1);
}
console.log(l.join(" "));